docs/contributing/code-review.md

```diff
@@ -42,6 +42,58 @@ Here are some guidelines for merging and reviewing in this case:
   make sure you push a PR with the change *and merge it quickly* to make sure that the changes
   are persisted across future deploys. Self merging is acceptable here, although this general
   class of changes should be limited as much as possible.
+  
+## Self-merging as a community partner
+
+As part of our [shared responsibility model](https://docs.2i2c.org/en/latest/about/service/shared-responsibility.html), we may grant merge rights to partner engineers.
+This allows others to merge changes that impact their community's infrastructure without requiring intervention from a 2i2c engineer.
+
+Merge rights are only given for partners with which we have built a relationship of trust, on a case-by-case basis.
+2i2c ultimately retains responsibility for the configuration of this infrastructure and its configuration.
+
+[See the `2i2c-org/collaborators` team](https://github.com/orgs/2i2c-org/teams/collaborators) for a list of sub-teams and individuals with write access to this repository.
+
+### Guidelines for community partners
+
+Our goal is to provide trusted communities the ability to more quickly make changes to their infrastructure in order to lead to a better, more collaborative service for their community.
+However, merge rights are a big responsibility, so please be careful in your actions.
+
+Community partners *may* self-merge if they want to, provided the following conditions are met:
+
+1. **They are confident debugging any issues that arise from the issue**.
+   If any issues arise from your self-merge, **you are responsible for resolving them, or reverting the change**.
+   You should understand the potential repurcussions of your change, and be ready to fix things if they break.
+2. **They have access to their cloud cluster to debug changes**.
+   Sometimes an issue requires intervention directly in the cloud infrastructure.
+   Before self-merging, ensure that you are authorized and have login-access to this infrastructure.
+3. **The change *only* touches files inside the `config/clusters/<cluster-name>` directory for their cluster.**
```

Comment:
We should be able to have fine-grained access to their own files with [codeowners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) in the future, right? For now, this advice is OK-ish.

Reply:
No, all codeowners does is automatically requests reviews from those listed in the codeowners file https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#about-code-owners

Reply:
Yeah, my previous sentence was a wish to have fine-grained permission through codeowners in the future...

Reply:
I don't know if that is on GitHub's roadmap, but I don't believe it is achievable right now. It is only about assigning reviewers. So you _can_ assign specific reviewers to PRs that affect a specific filepath. But you _cannot_ grant _write access_ to only a certain filepath within the repo.